autoscale/src/main/java/com/sequenceiq/periscope/controller/AutoScaleClusterCommonService.java

```diff
@@ -122,6 +129,17 @@ public Cluster setStopStartScalingState(Long clusterId, Boolean enableStopStartS
         return cluster;
     }
 
+    public Boolean shouldEnableStopStartBasedOnEntitlement(Cluster cluster, Integer timeAlertsIfPresent, Boolean enableStopStartScaling) {
+        if (enableStopStartScaling == null && timeAlertsIfPresent.equals(0)) {
+            if (entitlementValidationService.stopStartAutoscalingEntitlementEnabled(ThreadBasedUserCrnProvider.getAccountId(), cluster.getCloudPlatform())) {
+                enableStopStartScaling = true;
+            } else {
+                enableStopStartScaling = false;
+            }
+        }
+        return enableStopStartScaling;
+    }
```

Comment:
Nit: Since this method has two different invocations, you can overload them for multiple usages, and change the up the logic to use `isEmpty`, instead of `timeAlerts.equals(0)`

Reply:
Done